Jenkinsfile

```diff
@@ -0,0 +1,91 @@
+
+        MAVEN_MIRROR_REPOSITORY = "http://nexus3-kogito-tools.apps.kogito.automation.rhmw.io/repository/maven-public"
+    }
+    stages {
+        stage('Clean Workspace') {
+            steps {
+                dir ("${WORKING_DIR}") {
+                    deleteDir()
+                }
+            }
+        }
+        stage('Clone repository') {
+            steps {
+              sh "mkdir -p ${WORKING_DIR}"
+              sh "git clone https://github.com/kiegroup/kogito-cloud-operator.git ${WORKING_DIR}"
+            }    
+        }
+        stage('Initialize') {
+            steps {
+                sh "set +x && oc login --token=\$(oc whoami -t) --server=${OPENSHIFT_API} --insecure-skip-tls-verify"
+            }
+        }
+        stage('Build Kogito Operator') {
+            steps {
+                dir ("${WORKING_DIR}") {
+                    sh """
+                        export GOPATH=/home/jenkins/go
+                        export GOCACHE=\${GOPATH}/.cache/go-build
+                        export GOROOT=`go env GOROOT`
+                        GO111MODULE=on 
+                        go get -u golang.org/x/lint/golint
+                        touch /etc/sub{u,g}id
+                        usermod --add-subuids 10000-75535 \$(whoami)
+                        usermod --add-subgids 10000-75535 \$(whoami)
+                        cat /etc/subuid
+                        cat /etc/subgid
+                        make image_builder=podman
+                    """
+                }
+            }
+            
+        }
+        stage('Build Kogito CLI') {
+            steps {
+                dir ("${WORKING_DIR}") {
+                    sh "make build-cli"
+                }
+            }
+        }
+        stage('Push Operator Image to Openshift Registry') {
+            steps {
+                dir ("${WORKING_DIR}") {
+                    sh """
+                        set +x && podman login -u jenkins -p \$(oc whoami -t) --tls-verify=false ${OPENSHIFT_REGISTRY}
+                        cd version/ && TAG_OPERATOR=\$(grep -m 1 'Version =' version.go) && TAG_OPERATOR=\$(echo \${TAG_OPERATOR#*=} | tr -d '"')
+                        podman tag quay.io/kiegroup/kogito-cloud-operator:\${TAG_OPERATOR} ${OPENSHIFT_REGISTRY}/openshift/kogito-cloud-operator:pr-\$(echo \${GIT_COMMIT} | cut -c1-7)
+                        podman push --tls-verify=false docker://${OPENSHIFT_REGISTRY}/openshift/kogito-cloud-operator:pr-\$(echo \${GIT_COMMIT} | cut -c1-7)
+                    """
+                }
+            }
+        }
+        stage('Running Smoke Testing') {
+            steps {
+                dir ("${WORKING_DIR}") {
+                    sh """
+                        make run-smoke-tests operator_image=${OPENSHIFT_REGISTRY}/openshift/kogito-cloud-operator operator_tag=pr-\$(echo \${GIT_COMMIT} | cut -c1-7) maven_mirror=${MAVEN_MIRROR_REPOSITORY} concurrent=3
```

Comment:
From my personal experience there may be issues with running several concurrent tests with operator image pushed into internal OCP registry. If several tests started at the same time and pulled image from openshift namespace then sometimes the pull failed (with unauthorized error).
Have you experienced similar behaviour?

Reply:
Related, I'm seeing some `cucumber-*` namespaces in the AWS instance. Side effect?

Reply:
The namespaces should be deleted even in case of failure.
Maybe it is a leftover from manual testing?

Reply:
It seems that sometimes the namespace is not removed ... I still haven't been able to identify what causes it.

Reply:
Could it be that you stop the pipeline manually ?

Reply:
Make sense! I will test this.

Reply:
@radtriste I guess you're right. If the pipeline is stopped manually, the namespace will not be removed.